plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -195,16 +199,18 @@ processTerm = handleTerm <=< traverseOf termSubtypes applyTypeSubstitution where
             pure $ mkLet a NonRec bs' t'
         -- This includes recursive let terms, we don't even consider inlining them at the moment
         t -> forMOf termSubterms t processTerm
-    applyTypeSubstitution :: Type tyname uni a -> m (Type tyname uni a)
-    applyTypeSubstitution = typeSubstTyNamesM substTyName
+    applyTypeSubstitution :: Type tyname uni a -> Inlining tyname name uni fun a (Type tyname uni a)
+    applyTypeSubstitution t = gets isTypeEmpty >>= \case
```

Comment:
add a comment! "The type substitution is very often empty, and there are lots of types in the program, so this saves a lot of work (determined from profiling)"